iOS/Flows/Conversation/Cells/ConversationMessagesCell.swift

```diff
@@ -70,90 +62,34 @@ class ConversationMessagesCell: UICollectionViewCell {
         fatalError("init(coder:) has not been implemented")
     }
 
-    /// If true we need scroll to the most recent item upon layout.
-    private var needsOffsetReload = true
-
     override func layoutSubviews() {
         super.layoutSubviews()
 
         self.collectionView.expandToSuperviewSize()
-        self.collectionLayout.invalidateLayout()
-        self.collectionLayout.prepare()
-
-        if self.needsOffsetReload,
-           let contentOffset = self.collectionLayout.getMostRecentItemContentOffset() {
-
-            self.collectionView.contentOffset = contentOffset
-            self.collectionLayout.invalidateLayout()
-
-            self.needsOffsetReload = false
-        }
     }
 
     /// Configures the cell to display the given messages. The message sequence should be ordered newest to oldest.
     func set(sequence: MessageSequence) {
         self.conversation = sequence
 
-        // Separate the user messages from other message.
```

Comment:
All of this is moved to the datasource. Thread and conversation data source behavior is largely the same (except that threads only have one section, I'll be doing that in the next PR).